src/app/history/page.tsx

```diff
@@ -147,19 +153,57 @@ function TransactionHistoryContent() {
   const handleAccountClick = async (account: GroupedAccount) => {
     setSelectedAccount(account);
     setIsFetchingHistory(true);
+    setTransactionHistory({ data: [], nextPage: null });
 
     try {
       const history = await getAccountHistory(account.chainId, account.address);
-      setTransactionHistory(history);
+      if (history) {
+        setTransactionHistory({
+          data: history.transactions,
+          nextPage: history.pagination?.nextPage || null,
+        });
+      }
     } catch (error) {
       console.error("Error fetching transaction history:", error);
-      setTransactionHistory(null);
     } finally {
       setIsFetchingHistory(false);
     }
   };
 
-  // Add state for formatted amounts
+  const transactionListRef = useRef<HTMLDivElement>(null);
+
+  const handleLoadMore = async () => {
```

Comment:
You should use [useCallback](https://react.dev/reference/react/useCallback) for this function